src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -999,6 +999,130 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             return this;
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
+
+            if (value != null && value.Length > 0)
+            {
+                if (value != this)
```

Comment:
@AlexGhiondea I complete understand that you write about.
If you changed next step  `Thread 1: Call sb1.Append(sb2)` on `Thread 1: Call sb1.Append(sb2.ToString())` nothing is changed because 
`Thread 2: Call sb2.Replace("a", "B")` will produce thread racing with `sb2.ToString()` code.